File: app/src/main/java/com/heejong/githubAPI_kotlin/network/APIInterface.java
package com.heejong.githubAPI_kotlin.network

import com.heejong.githubAPI_kotlin.getGithubInfomation
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Query


internal interface APIInterface {

    @GET("")
    fun getAPIinfo(@Query("url") url: String): Call<getGithubInfomation>
    //?client_id={client_id}&client_secret={client_secret}&search={search}
}

File: app/src/main/java/com/heejong/githubAPI_kotlin/network/APIInterface.kt
package com.heejong.githubAPI_kotlin.network

import com.heejong.githubAPI_kotlin.getGithubInfomation
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Headers
import retrofit2.http.Path
import retrofit2.http.Query
import io.reactivex.Single


internal interface APIInterface {
    @GET("users/{username}")
    fun requestContributors(
        @Path("username") userName:String
        ) : Single<getGithubInfomation>

}